File: go/hamming/hamming.go
package hamming

import "errors"

const testVersion = 6

// Distance returns Hammock distance between DNA strands.
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("Strings lengths are different")
	}
	distance := 0
	for i := range a {
		if a[i] != b[i] {
			distance++
		}
	}
	return distance, nil
}

File: clojure/hamming/src/hamming.clj
(ns hamming)

(defn distance-value
  "Return 0 if given elements are equal, 1 otherwise"
  [a b]
  (if (= a b) 0 1))

(defn distance
  "Returns Hamming distance between given DNA strands"
  [dna-left dna-right]
  (when (= (count dna-left) (count dna-right))
    (reduce + (map distance-value dna-left dna-right))))
